 syntax = "proto3";

package routesia.config;

import "routesia/dhcp/dhcp.proto";
import "routesia/dns/cache/cache.proto";
import "routesia/dns/authoritative/authoritative.proto";
import "routesia/interface/interface.proto";
import "routesia/ipam/ipam.proto";
import "routesia/netfilter/netfilter.proto";
import "routesia/route/route.proto";


message Config {
    // System schema
    //
    message System {
        string schema = 1;
        int64 version = 2;
    }
    System system = 1;

    // IP address management module
    //
    routesia.ipam.IPAMConfig ipam = 2;

    // Interface module
    //
    routesia.interface.InterfaceModuleConfig interface = 3;

    // Route module
    //
    routesia.route.RouteModuleConfig route = 4;

    // Netfilter module
    //
    routesia.netfilter.NetfilterConfig netfilter = 5;

    // DHCP module
    //
    routesia.dhcp.DHCPConfig dhcp = 6;

    // DNS cache module
    //
    routesia.dns.cache.DNSCacheConfig dns_cache = 7;

    // Authoritative DNS module
    //
    routesia.dns.authoritative.AuthoritativeDNSConfig authoritative_dns = 8;
}


message CommitResult {
    // Result codes
    //
    enum CommitResultCode {
        COMMIT_SUCCESS = 0;
        COMMIT_UNCHANGED = 1;
        COMMIT_ERROR = 2;
    }
    CommitResultCode result_code = 1;

    // Extra message as a string
    //
    string message = 2;
}
